File: go/fiber/internal/modules/branch/module.go
package branch

import (
	"database/sql"
	auth "tcc/internal/modules/auth/middleware"

	"github.com/gofiber/fiber/v2"
)

func BranchModule(router *fiber.App, db *sql.DB) {
	repository := BranchRepository{
		db: db,
	}
	service := BranchService{
		repository: repository,
	}
	controller := BranchController{
		service: service,
	}

	group := router.Group("/agencias")
	{
		group.Get("", auth.AuthMiddleware(), controller.FindAll)
	}
}

File: rust/axum/src/modules/branch/module.rs
use axum::{Router, routing::get, middleware::from_fn,};
use crate::modules::auth::middleware::auth_middleware;
use super::controller::BranchController;

pub fn branch_module() -> Router {
    let router = 
    Router::new()
    .route("/", get(BranchController::find_all)).layer(from_fn(auth_middleware));

    Router::new().nest("/agencias", router)
}
